solarforecastarbiter/datamodel.py

```diff
@@ -1230,10 +1233,72 @@ def __post_init__(self):
                 'One of observation_id OR aggregate_id must be set')
 
 
+def __check_plot_spec__(plot_spec):
+    """Ensure that the provided plot specification is a valid JSON object"""
+    try:
+        spec_dict = json.loads(plot_spec)
+        validate(instance=spec_dict, schema={'type': 'object'})
+    except (json.JSONDecodeError, ValidationError):
+        raise ValueError('Figure spec must be a valid json object.')
+
+
 @dataclass(frozen=True)
 class ReportFigure(BaseModel):
+    """Parent class for different types of Report Figures"""
+    def __post_init__(self):
+        if type(self) == ReportFigure:
+            raise ValueError("Invalid Report Figure. Figures must be of class "
+                             "PlotlyReportFigure or BokehReportFigure.")
+
+    @classmethod
+    def from_dict(model, input_dict, raise_on_extra=False):
+        dict_ = input_dict.copy()
+        if model != ReportFigure:
+            return super().from_dict(dict_, raise_on_extra)
+        figure_class = dict_.get('figure_class')
+        if figure_class == 'plotly':
+            return PlotlyReportFigure.from_dict(dict_, raise_on_extra)
+        elif figure_class == 'bokeh':
+            return BokehReportFigure.from_dict(dict_, raise_on_extra)
+        else:
+            raise NotImplementedError(
+                f'Do not know how to process dict into a ReportFigure.')
+
+
+@dataclass(frozen=True)
+class PlotlyReportFigure(ReportFigure):
     """A class for storing metric plots for a report with associated metadata.
 
+    Parameters
+    ----------
+    name: str
+        A descriptive name for the figure.
+    spec: str
+        JSON string representation of the plotly plot.
+    svg: str
+        A static svg copy of the plot, for including in the pdf version.
+    figure_type: str
+        The type of plot, e.g. bar or scatter.
+    category: str
+        The metric category. One of ALLOWED_CATEGORIES keys.
+    metric: str
+        The metric being plotted.
+    """
+    name: str
+    spec: str
+    svg: str
+    figure_type: str
+    figure_class: str = 'plotly'
```

Comment:
Move this last and possibly make it a ``field(init=False)`` so users can't set it